forms/core/Application.php

```diff
@@ -119,98 +113,63 @@ public static function authenticate()
             }
         }
 
-
-        // do I have an authorization token
         $authorizationToken = Settings::get('access_token');
 
+        if (!empty($error)) {
+            $description = Request::get('error_description');
 
-        if (!empty($authorizationToken)){

```

Comment:
Basically, the main change in the file was removing this part isn't it (and ensuring that when storing the expiry, also added the current time too).